src/Components/Server/src/Circuits/CircuitHost.cs

```diff
@@ -24,6 +25,7 @@ internal class CircuitHost : IAsyncDisposable
         private readonly ILogger _logger;
         private bool _initialized;
         private bool _disposed;
+        private JsonSourceGeneration.JsonContext _jsonContext;
```

Comment:
In `JSInteropMethods`, you made the JsonContext `static`, but here and in `IpcReceiver` it is an instance based field. Was there a reason for the difference?

Reply:
@eerhardt Are we allowed to treat `JsonSourceGeneration.JsonContext` as thread-safe?

Possibly that's the reason for the distinction but I'm not sure. @pranavkm can confirm!

Reply:
Oh actually I think I see why it's done like this.

Each circuit has its own distinct `ElementReferenceJsonConverter` instance because it's hooked up to some state within the circuit (a circuit-local sequence of IDs so you don't disclose info about other circuits). This means the JSON serialization options are per-circuit, which in turn means the JSON context is now per-circuit.

Reply:
Also note that `JSInteropMethods` is within the WebAssembly project, so it's not part of the Blazor Server circuits concept and can freely make things `static` (it's a single-user environment that doesn't have to hide info about other users).

Reply:
> Are we allowed to treat JsonSourceGeneration.JsonContext as thread-safe?

The intent is that it is thread-safe & any issues here would be addressed.

Reply:
Yeah, this isn't about thread safety. Like @SteveSandersonMS said, the JsonSerializerOptions are circuit-scoped in Blazor server, consequently the context has to be too.

Reply:
Makes sense. Thanks for the explanation.